elide-core/src/test/java/com/yahoo/elide/core/EntityDictionaryTest.java

```diff
@@ -486,14 +487,156 @@ public void testChildBinding() {
+
+        this.bindEntity(SuperclassBinding.class);
+
+        assertEquals(SuperclassBinding.class, getEntityBinding(SubclassBinding.class).entityClass);
+        assertEquals(SuperclassBinding.class, getEntityBinding(SuperclassBinding.class).entityClass);
+
+        assertEquals(SuperclassBinding.class, lookupIncludeClass(SuperclassBinding.class));
+        assertEquals(SuperclassBinding.class, lookupIncludeClass(SubclassBinding.class));
+        assertEquals(SuperclassBinding.class, lookupIncludeClass(SubsubclassBinding.class));
+    }
+
+    @Test
+    public void testNonEntityInheritanceBindingOverride() {
+        @Include
+        class SuperclassBinding {
+            @Id
+            private long id;
+        }
+
+        class SubclassBinding extends SuperclassBinding {
+        }
+
+        @Include
+        class SubsubclassBinding extends SubclassBinding {
+            @Id
+            private long id;
+        }
+
+        this.bindEntity(SuperclassBinding.class);
+        this.bindEntity(SubsubclassBinding.class);
+
+        assertEquals(SuperclassBinding.class, getEntityBinding(SubclassBinding.class).entityClass);
+        assertEquals(SuperclassBinding.class, getEntityBinding(SuperclassBinding.class).entityClass);
+        assertEquals(SubsubclassBinding.class, getEntityBinding(SubsubclassBinding.class).entityClass);
+
+        assertEquals(SuperclassBinding.class, lookupIncludeClass(SuperclassBinding.class));
+        assertEquals(SuperclassBinding.class, lookupIncludeClass(SubclassBinding.class));
+        assertEquals(SubsubclassBinding.class, lookupIncludeClass(SubsubclassBinding.class));
+    }
+
+    @Test
+    public void testNonEntityInheritanceBindingExclusion() {
+        @Include
+        class SuperclassBinding {
+            @Id
+            private long id;
+        }
+
+        class SubclassBinding extends SuperclassBinding {
+        }
+
+        @Exclude
+        class SubsubclassBinding extends SubclassBinding {
+        }
+
+        this.bindEntity(SuperclassBinding.class);
+        this.bindEntity(SubsubclassBinding.class);
+
+        assertEquals(SuperclassBinding.class, getEntityBinding(SubclassBinding.class).entityClass);
+        assertEquals(SuperclassBinding.class, getEntityBinding(SuperclassBinding.class).entityClass);
+        assertThrows(IllegalArgumentException.class, () -> { getEntityBinding(SubsubclassBinding.class); });
+
+        assertEquals(SuperclassBinding.class, lookupIncludeClass(SuperclassBinding.class));
+        assertEquals(SuperclassBinding.class, lookupIncludeClass(SubclassBinding.class));
+        assertEquals(null, lookupIncludeClass(SubsubclassBinding.class));
+    }
+
+    @Test
+    public void testGetFirstAnnotation() {
+        @Exclude
+        class Foo { }
+
+        @Include
+        class Bar extends Foo {
+
+        }
+
+        class Baz extends Bar {
+
+        }
+
+        Annotation first = getFirstAnnotation(Baz.class, Arrays.asList(Exclude.class, Include.class));
```

Comment:
This case should not happen, but do we need to check the case that `@Include` and `@Exclude` are declared on the same class together?